### PR TITLE
Add item group record support for invoices

### PR DESCRIPTION
**Pull request by SanduDS**
# Description

This is to improve the invoice support and to add itemgroup records for the connector

Fixes #https://github.com/ballerina-platform/ballerina-extended-library/issues/473

One line release note: 
- Improved the netsuite invoice record CRUD operations with item-group record support
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X]  Unit tests 
- [X] Intergation tests

**Test Configuration**:
* Ballerina Version: Ballerina 2201.2.1 (Swan Lake Update 2)
* Operating System: Ubuntu
* Java SDK: JAVA 11

# Checklist:

### Security checks
 - [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: netsuite/Invoice.bal [hunk 1]
```diff
@@ -113,11 +113,44 @@ isolated function mapInvoiceRecord(xml response) returns Invoice|error {
...
+            amount: extractDecimalFromXML(elementItem/**/<tranSales:amount>/*),  
+            location: extractRecordRefFromXML(elementItem/**/<tranSales:location>/*),
+            line: extractDecimalFromXML(elementItem/**/<tranSales:line>/*)
+        };
+        boolean|error value = extractBooleanValueFromXMLOrText(element/**/<tranSales:excludeFromRateRequest>/*);
+        if (value is boolean) {
+            itemMember.excludeFromRateRequest = value;
+        }
+        value = extractBooleanValueFromXMLOrText(element/**/<tranSales:isTaxable>/*);
+        if (value is boolean) {
```
Comment (LakshanSS) [inline netsuite/Invoice.bal:146]:
```suggestion
        if value is boolean {
            itemMember.excludeFromRateRequest = value;
        }
        value = extractBooleanValueFromXMLOrText(element/**/<tranSales:isTaxable>/*);
        if value is boolean {
```

File: netsuite/account.bal [hunk 2]
```diff
@@ -54,6 +54,143 @@ isolated function mapNewAccountRecordFields(NewAccount account) returns string {
     return finalResult;
 }
 
+isolated function mapNewItemGroupRecordFields(NewItemGroup itemGroup) returns string|error {
+    string finalResult = EMPTY_STRING;
+    map<anydata>|error itemGroupMap = itemGroup.cloneWithType(MapAnyData);
+    if (itemGroupMap is map<anydata>) {
```
Comment (LakshanSS) [inline netsuite/account.bal:60]:
```suggestion
    if itemGroupMap is map<anydata> {
```

File: netsuite/account.bal [hunk 3]
```diff
@@ -54,6 +54,143 @@ isolated function mapNewAccountRecordFields(NewAccount account) returns string {
...
+        string[] keys = itemGroupMap.keys();
+        int position = 0;
+        foreach var item in itemGroup {
+            if (item is string|boolean|decimal) {
```
Comment (LakshanSS) [inline netsuite/account.bal:64]: shall we remove these unnecessary `()` in all the places?

**Review by LakshanSS — COMMENTED** [review]
Added minor comments

**Comment by SanduDS** [comment]
@LakshanSS Discussed changes were done

**Review by LakshanSS — APPROVED** [review]

**Pull request merged**